### PR TITLE
Extend the logger formatter instead of overwriting it 

### PR DESCRIPTION
**Pull request by wshihadeh**
WIP
Extend the logger formatter instead of overwriting it

### CODE REVIEW

File: lib/request_headers_logger/message_queue/message_queue_plugin.rb [hunk 1]
```diff
@@ -25,18 +25,19 @@ def symbolize(obj = {})
         store = symbolize(properties.headers).dig(:store) || {}
```
Comment (tonchev-ivan) [inline lib/request_headers_logger/message_queue/message_queue_plugin.rb:25]: please change it to safe call `&.dig(:store)`

**Comment by tonchev-ivan** [comment]
👍

File: lib/request_headers_logger/text_formatter.rb [hunk 2]
```diff
@@ -1,9 +1,17 @@
...
+
 module RequestHeadersLogger
-  class TextFormatter < ::Logger::Formatter
-    def call(severity, timestamp, progname, msg)
-      super(severity, timestamp, progname, "#{tags_text}#{msg}")
+  module TextFormatter
+    def call(severity, time, progname, msg)
+      format(::Logger::Formatter::Format,
+             severity[0..0],
+             format_datetime(time),
```
Comment (csmuc) [inline lib/request_headers_logger/text_formatter.rb:10]: The presence of the `format_datetime' method is not a safe assumption. For instance the formatter might by just a Proc

File: lib/request_headers_logger/message_queue/message_queue_plugin.rb [hunk 3]
```diff
@@ -25,18 +25,19 @@ def symbolize(obj = {})
...
       lifecycle.after(:consume) do |delivery_info, properties, payload|
         RequestHeadersMiddleware.store = {}
       end
     end
 
-    def self.set_dj_loggers
+    def self.set_mq_loggers
       RequestHeadersLogger.configure do |config|
-        config[:loggers] = [MessageQueue.logger]
+        config[:loggers] << MessageQueue.logger
```
Comment (csmuc) [inline lib/request_headers_logger/message_queue/message_queue_plugin.rb:38]: memory leak if I read the code correctly (this code hits on every message delivery right?)

File: lib/request_headers_logger/delayed_job/request_header_delayed_plugin.rb [hunk 4]
```diff
@@ -26,8 +26,9 @@ def self.set_dj_loggers
         loggers = [Delayed::Worker.logger]
         loggers << ::Rails.logger
 
-        config[:loggers] = loggers
+        config[:loggers].push(loggers).flatten!.uniq!
       end
+      RequestHeadersLogger.prepare_loggers
```
Comment (csmuc) [inline lib/request_headers_logger/delayed_job/request_header_delayed_plugin.rb:31]: looks like an expensive operation to me, runs for every job right?

**Pull request merged**